### DISCUSSION TITLE
Valgrind automatic execution in C++

### DISCUSSION THREAD

**Discussion by j-balkovec** [community/community#55256] [Programming Help]
### Body

Hi, 

I am working on a school project with a Covid DataBase represented as a Hash Map. I am pretty much done with the assignment all I want to add is to be able to execute a `valgrind` command from the "menu"

```
/**
 * @brief Displays the menu for user interaction.
 */
void menu() {
  std::cout << "\n*** Welcome to the COVID-19 Data Base ***\n\n"
            << "[1]\tCreate a Hash Table with the WHO file\n"
            << "[2]\tAdd a Data entry\n"
            << "[3]\tGet a Data entry\n"
            << "[4]\tRemove a Data entry\n"
            << "[5]\tDisplay HasH Table\n"
            << "[6]\tLog Data [covid_db.log]\n"
            << "[7]\tLog Memory [valgrind.log]\n"
            << "[8]\tQuit\n";
}
```

As you can see I am working on #7. I don't have a lot of experience with using/passing command line arguments besides argv argc*[] in main. I did some research and I found some code online that uses these libraries:

```
#include <unistd.h>
#include <sys/types.h>
#include <sys/wait.h>
```

I am honestly scared to use them since I've set off a fork bomb on the school's server before. I was using multithreading and apparently, the code was so bad it sent/ran a bunch of child processes which made the server crash...

Here is what I've got so far, pretty basic stuff:

```
void CovidDB::log_memory() {
  #ifdef LOG
  std::string command = "valgrind --leak-check=full --show-leak-kinds=all --track-origins=yes --num-callers=20 --log-file=valgrind.log ./main";
  std::system(command.c_str());

  log(); //logs data to a covid_db.log file

  std::exit(EXIT_SUCCESS);
}
#endif
```

The only problem I have is that when the user selects 7 it re-run the program and the user has to exit it on their own in order to get the report. Is there a way to avoid that? (By that I mean to make it fully automatic)

Thank you.
### Guidelines

- [x] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by airtower-luna** [comment]
> I am pretty much done with the assignment all I want to add is to be able to execute a valgrind command from the "menu"

What exactly do you want to do with Valgrind? Usually you'd run your program from Valgrind to check for memory issues, not the other way around. :thinking:
